scipy/signal/_filter_design.py

```diff
@@ -4218,15 +4240,22 @@ def buttord(wp, ws, gpass, gstop, analog=False, fs=None):
     >>> plt.show()
 
     """
+    xp = array_namespace(wp, ws)
+    wp, ws = map(xp.asarray, (wp, ws))
+
     _validate_gpass_gstop(gpass, gstop)
     fs = _validate_fs(fs, allow_none=True)
-    wp, ws, filter_type = _validate_wp_ws(wp, ws, fs, analog)
-    passb, stopb = _pre_warp(wp, ws, analog)
-    nat, passb = _find_nat_freq(stopb, passb, gpass, gstop, filter_type, 'butter')
+    wp, ws, filter_type = _validate_wp_ws(wp, ws, fs, analog, xp=xp)
+    passb, stopb = _pre_warp(wp, ws, analog, xp=xp)
+    nat, passb = _find_nat_freq(
+        stopb, passb, gpass, gstop, filter_type, 'butter', xp=xp
+    )
 
-    GSTOP = 10 ** (0.1 * abs(gstop))
-    GPASS = 10 ** (0.1 * abs(gpass))
-    ord = int(ceil(log10((GSTOP - 1.0) / (GPASS - 1.0)) / (2 * log10(nat))))
+    GSTOP = 10 ** (0.1 * builtins.abs(gstop))
```

Comment:
How come we have to specify `builtins.abs`, we haven't imported any other `abs`s?

Reply:
Unfortunately https://github.com/scipy/scipy/pull/22911/files#diff-7adae798f824ec3b9bd0ac54c06124831ceccbea4b7d578d6d71f292e9a25b3cL43